src/smif/convert/area.py

```diff
@@ -120,113 +127,12 @@ def __len__(self):
-
-        """
-        if name not in self._register:
-            raise ValueError("Region set '{}' not registered".format(name))
-        return self._register[name]
-
-    def register(self, region_set):
-        """Register a set of regions as a source/target for conversion
-        """
-        if region_set.name in self._register:
-            msg = "A region set named {} has already been loaded"
-            raise ValueError(msg.format(region_set.name))
-
-        already_registered = self.names
-        self._register[region_set.name] = region_set
-        for other_set_name in already_registered:
-            self._generate_coefficients(region_set, self._register[other_set_name])
-
-    def convert(self, data, from_set_name, to_set_name):
-        """Convert a list of data points for a given set of regions
-        to another set of regions.
-
-        Parameters
-        ----------
-        data: numpy.ndarray with dimension regions
-        from_set_name: str
-        to_set_name: str
-
-        """
-        from_set = self.get_entry(from_set_name)
-        from_set_names = from_set.get_entry_names()
-        to_set = self.get_entry(to_set_name)
-        to_set_names = to_set.get_entry_names()
-
-        converted = np.zeros(len(to_set))
-        coefficents = self._conversions[from_set.name][to_set.name]
-
-        for from_region_name, from_value in zip(from_set_names, data):
-            for to_region_name, coef in coefficents[from_region_name]:
-                to_region_idx = to_set_names.index(to_region_name)
-                converted[to_region_idx] += coef*from_value
-
-        return converted
-
-    def _generate_coefficients(self, set_a, set_b):
-        msg = "Generating region coefficients from set %s to %s"
-        self.logger.info(msg, set_a, set_b)
-        # from a to b
-        self._conversions[set_a.name][set_b.name] = self._conversion_coefficients(set_a, set_b)
-        # from b to a
-        self._conversions[set_b.name][set_a.name] = self._conversion_coefficients(set_b, set_a)
-
-    @staticmethod
-    def _conversion_coefficients(from_set, to_set):
-        """Return a dict containing the proportions of to_regions intersecting
-        with each given from_region::
-
-            {
-                from_region.name: [
-                    (to_region.name, proportion),
-                    # ...
-                ],
-                # ...
-            }
-
-        """
-        coefficients = defaultdict(list)
-
-        for to_region in to_set:
-            intersecting_from_regions = from_set.intersection(to_region.shape.bounds)
-
-            for from_region in intersecting_from_regions:
-                proportion = proportion_of_a_intersecting_b(from_region.shape, to_region.shape)
-                coefficient_pair = (to_region.name, proportion)
-                coefficients[from_region.name].append(coefficient_pair)
-
-        return coefficients
+        super().__init__(axis=0)
```

Comment:
`axis` looks important in any `Register`, as it states the canonical shape of the data it can handle. Maybe worth an extra comment in each of `RegionRegister` and `IntervalRegister`.